File: common-rule-engine/src/test/files/checks/DoublePrefixOperatorCheck/DoublePrefixOperatorCheck.go
package DoublePrefixOperatorCheck
func test1() {
  var a int = 1;
  var flag bool = true;
  var hash int = 1;
  var hashCode int = 2;

  var a1 int = ^^^a; // Noncompliant {{Use the "^" operator just once or not at all.}}
  var flag2 bool  = !!flag;  // Noncompliant
  var flag3 bool  = !!!flag; // Noncompliant
  var flag4 bool  = !!!foo(); // Noncompliant
  var flag5 bool  = !(!flag4); // Noncompliant  {{Use the "!" operator just once or not at all.}}
  var flag6 bool  = !(!(!flag4)); // Noncompliant {{Use the "!" operator just once or not at all.}}
  var c int = ^(^(^a3));    // Noncompliant  {{Use the "^" operator just once or not at all.}}

  var a3  int =  - - -a2;  // Noncompliant
  var a4  int =  - -a2;  // Noncompliant
  var a5  int =  + + +a2;  // Noncompliant
  var a6  int =  + +a2;  // Noncompliant
  var a2  int = ^^a; // Noncompliant
  var a9  int = ^(^a); // Noncompliant
  var a10 int  = ^(^(a9 - 9));  // Noncompliant

  var flag1 bool = !flag4;  // Compliant
  var b =int   ^a1;  // Compliant
  var flag7 bool = !foo(); // Compliant
  var h1 int  = ^^hash; // Noncompliant
  var h2 int  = ^^hashCodep; // Noncompliant
}

File: common-rule-engine/src/test/files/checks/DoublePrefixOperatorCheck/DoublePrefixOperatorCheck.java
class test1 {
  int a = 1;
  boolean flag = true;
  int hash = 1;
  int hashCode = 2;

  boolean flag2 = !!flag;  // Noncompliant
  int a1 = ~~~a; // Noncompliant {{Use the "~" operator just once or not at all.}}

  boolean flag3 = !!!flag; // Noncompliant

  boolean flag4 = !!!foo(); // Noncompliant 

  boolean flag5 = !(!flag4); // Noncompliant  {{Use the "!" operator just once or not at all.}}

  int c = ~(~(~a3));    // Noncompliant  {{Use the "~" operator just once or not at all.}}

  boolean flag6 = !(!(!flag4)); // Noncompliant {{Use the "!" operator just once or not at all.}}

  int a3 =  - - -a2;  // Noncompliant 

  int a4 =  - -a2;  // Noncompliant 

  int a5 =  + + +a2;  // Noncompliant 

  int a6 =  + +a2;  // Noncompliant 

  int a2 = ~~a; // Noncompliant

  int a9 = ~(~a); // Noncompliant

  int a10 = ~(~(a9 - 9));  // Noncompliant

  int a7 = --a2;  // Compliant
  int a8 = ++a2;  // Compliant
  boolean flag1 = !flag4;  // Compliant
  int b = ~a1;  // Compliant
  boolean flag7 = !foo(); // Compliant
  int h1 = ~~hash; // Noncompliant
  int h2 = ~~hashCodep; // Noncompliant
}
